lte/gateway/c/session_manager/ShardTracker.cpp

```diff
@@ -13,45 +13,47 @@
 
 #include <string>
 #include <vector>
+#include <set>
 
 #include "ShardTracker.h"
 
 namespace magma {
 
-int ShardTracker::add_ue() {
-  /*
-  * If there are no shards, add an entry for the shard
-  * representing a single UE at the new shard
-  */
-  if (ue_count_per_shard_.size() == 0) {
-    ue_count_per_shard_.push_back(1);
-    return 0;
+uint16_t ShardTracker::add_ue(std::string imsi) {
+  if (imsis_per_shard_.empty()) {
+    std::set<std::string> new_shard = {imsi};
+    imsis_per_shard_.push_back(new_shard);
   }
-  /*
-  * Iterate through all existing shards, if all are full
-  * create a new shard with quantity 1, otherwise increment
-  * the quantity of UEs in the latest shard
-  */
-  for (size_t shard_id = 0; shard_id < ue_count_per_shard_.size(); shard_id++) {
-    if (ue_count_per_shard_[shard_id] < max_shard_size_) {
-      ue_count_per_shard_[shard_id]++;
+  for (size_t shard_id = 0; shard_id < imsis_per_shard_.size(); shard_id++) {
+    // If the UE is already in the shard, return the shard id. This check
+    // is meant to avoid multiple sessions for a UE being assigned duplicate
+    // entries.
```

Comment:
Assigning a reference to the set a variable might make this cleaner to read.
Something like
```c++
std::set<std::string>& imsis = imsis_per_shard_[shard_id];
```